### PR TITLE
feat(New Relic): Add support for new relic

### PR DESCRIPTION
**Pull request by streibeb**

### CODE REVIEW

File: trust-provider/trust_provider.go [hunk 1]
```diff
@@ -366,10 +374,16 @@ func New(onboarding Onboarding, rules []Rule, account Account, resolver did.Reso
...
+	}
+
+	t.router.HandleFunc(applyNewRelic(app, "/api/register", t.register)).Methods("POST")
 
 	for _, rule := range rules {
-		t.router.HandleFunc(fmt.Sprintf("/api/%s/{token}", rule.Name), t.handleRule(rule)).Methods("POST")
+		t.router.HandleFunc(applyNewRelic(app, fmt.Sprintf("/api/%s/{token}", rule.Name), t.handleRule(rule))).Methods("POST")
 	}
 
 	http.Handle("/", handlers.LoggingHandler(os.Stdout, t.router))
```
Comment (lucastetreault) [inline trust-provider/trust_provider.go]:
```suggestion
	http.Handle(applyNewRelic(app, "/", handlers.LoggingHandler(os.Stdout, t.router)))
```

File: trust-provider/trust_provider.go [hunk 2]
```diff
@@ -366,10 +374,16 @@ func New(onboarding Onboarding, rules []Rule, account Account, resolver did.Reso
...
 	}
 
-	t.router.HandleFunc("/api/register", t.register).Methods("POST")
+	newRelicConfig := newrelic.NewConfig(os.Getenv("NEW_RELIC_APP_NAME"), os.Getenv("NEW_RELIC_LICENSE_KEY"))
+	app, err := newrelic.NewApplication(newRelicConfig)
+	if err != nil {
+		log.Println("Unable to create New Relic application:", err.Error())
+	}
+
+	t.router.HandleFunc(applyNewRelic(app, "/api/register", t.register)).Methods("POST")
```
Comment (lucastetreault) [inline trust-provider/trust_provider.go]:
```suggestion
	t.router.HandleFunc("/api/register", t.register).Methods("POST")
```

File: trust-provider/trust_provider.go [hunk 3]
```diff
@@ -366,10 +374,16 @@ func New(onboarding Onboarding, rules []Rule, account Account, resolver did.Reso
...
 
 	for _, rule := range rules {
-		t.router.HandleFunc(fmt.Sprintf("/api/%s/{token}", rule.Name), t.handleRule(rule)).Methods("POST")
+		t.router.HandleFunc(applyNewRelic(app, fmt.Sprintf("/api/%s/{token}", rule.Name), t.handleRule(rule))).Methods("POST")
```
Comment (lucastetreault) [inline trust-provider/trust_provider.go]:
```suggestion
		t.router.HandleFunc(fmt.Sprintf("/api/%s/{token}", rule.Name), t.handleRule(rule)).Methods("POST")
```

File: trust-provider/trust_provider.go [hunk 4]
```diff
@@ -366,10 +374,16 @@ func New(onboarding Onboarding, rules []Rule, account Account, resolver did.Reso
 		}
 	}
 
-	t.router.HandleFunc("/api/register", t.register).Methods("POST")
+	newRelicConfig := newrelic.NewConfig(os.Getenv("NEW_RELIC_APP_NAME"), os.Getenv("NEW_RELIC_LICENSE_KEY"))
```
Comment (lucastetreault) [inline trust-provider/trust_provider.go]: Can this stuff move in to the applyNewRelic function so that it doesn't clutter up the New function?

Comment (streibeb) [reply trust-provider/trust_provider.go]: I don't believe so, unless we want to create a new app for each endpoint created.

**Pull request merged**